docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,264 @@
+  <li>2</li> /* key: "2" */
+</ul>
+```
+
+In the event that the element containing "2" needs to be removed, React will hide the element instead of removing the object entirely like so:
+
+```html
+<ul>
+  <li>1</li> /* key: "1" */
+  <li style={display: 'none'}>2</li> /* key: "2" */
+</ul>
+```
+
+#### Example 2
+
+Given the html below:
+
+```html
+<ul>
+  <li>1</li> /* key: "1" */
+  <li>2</li> /* key: "2" */
+</ul>
+```
+
+In the event that the element "1" needs to be removed, React will swap the text and hide the second element like so:
+
+```html
+<ul>
+  <li>2</li> /* key: "2" */
+</ul>
+```
+
+## Conditional Rendering
+
+With React you can render distinct components depending on the value of your data. This means no longer needing to manually modify the CSS classes of HTML elements. Instead, you create distinct components that encapsulate behavior you need. Then, render only the specific component you need depending on the state of your application.
+
+This follows the concept of "separation of concerns" by allowing each component to manage how it should be rendered.
+
+### Element Variables
+
+Given the current state of your application you can assign elements as the value of variables, then only render the needed element.
+
+In the example below we have a stateful component named `<LoginControl />`. We use a boolean value `this.state.loggedIn` to track if the user is logged in. If logged in, we want to render `<LogoutButton />`. If logged out, we want to render `<LoginButton />`. Depending on the value of `this.state.loggedIn`, we use `loginButton` to store the element to be rendered.
+
+The highlighted lines below show where `loginControlButton` is created and set to a default value of "`<LoginButton />`", then where it conditionally gets changed depending on current state, and finally where `{loginControlButton}` is rendered.
+
+```javascript{21,23,27}
+function LogoutButton(props) {
+  return <button onClick={props.click}>Logout</button>
+}
+function LoginButton(props) {
+  return <button onClick={props.click}>Login</button>
+}
+class LoginControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {loggedIn: true}
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
+  }
+  login() {
+    this.setState({loggedIn: true,});
+  }
+  logout() {
+    this.setState({loggedIn: false});
+  }
+  render() {
+    var loginControlButton = <LoginButton click={this.login} />;
+    if (this.state.loggedIn) {
+      loginControlButton = <LogoutButton click={this.logout} />;
+    }
+    return (
+      <div>
+        {loginControlButton}
+      </div>
+    );
+  }
+}
+
+```
+
+<a target="_blank" href="https://codepen.io/ericnakagawa/pen/Egpdrz?editors=0010">Try it out on Codepen.</a>
+
+
+### Prevent Component Rendering
+
+In some cases, you will not want a component to render. To prevent components from rendering, return `null` or `false` from the `render()` function.
```

Comment:
React actually has a feature like this (you *can* return null from render) but this is not what you are describing. "Return null" means you actually return null. This is not what you are showing in the example. There, you include null instead of one of the children in the rendered output. Can you rephrase to make it clearer?